torchgeo/datasets/oscd.py

```diff
@@ -311,3 +314,190 @@ def get_masked(img: Tensor) -> Array:  # type: ignore[type-arg]
+            18976.0,
+            17874.0,
+            16611.0,
+            16512.0,
+            16394.0,
+            16672.0,
+            16141.0,
+            16097.0,
+            15716.0,  # (16097 + 15336)/2
+            15336.0,
+            15203.0,
+        ]
+    )
+
+    # min/max band statistics computed by percentile clipping the
+    # above to samples to [2, 98]
+    band_mins = torch.tensor(  # type: ignore[attr-defined]
+        [
+            1.0,
+            1.0,
+            1.0,
+            1.0,
+            1.0,
+            1.0,
+            1.0,
+            1.0,
+            1.0,
+            1.0,
+            1.0,
+            1.0,
+            1.0,
+        ]  # B10 = mode(all)
+    )
+    band_maxs = torch.tensor(  # type: ignore[attr-defined]
+        [
+            9859.0,
+            12872.0,
+            13163.0,
+            14445.0,
+            12477.0,
+            12563.0,
+            12289.0,
+            15596.0,
+            12183.0,
+            9458.0,
+            7677.0,  # (9458 + 5897)/2
+            5897.0,
+            5544.0,
+        ]
+    )
+
+    def __init__(
+        self,
+        root_dir: str,
+        bands: str = "all",
+        batch_size: int = 64,
+        num_workers: int = 0,
+        val_split_pct: float = 0.2,
+        **kwargs: Any,
+    ) -> None:
+        """Initialize a LightningDataModule for OSCD based DataLoaders.
+
+        Args:
+            root_dir: The ``root`` arugment to pass to the OSCD Dataset classes
+            bands: "rgb" or "all"
+            batch_size: The batch size to use in all created DataLoaders
+            num_workers: The number of workers to use in all created DataLoaders
+            val_split_pct: What percentage of the dataset to use as a validation set
+        """
+        super().__init__()  # type: ignore[no-untyped-call]
+        self.root_dir = root_dir
+        self.bands = bands
+        self.batch_size = batch_size
+        self.num_workers = num_workers
+        self.val_split_pct = val_split_pct
+
+        if bands == "rgb":
+            self.mins = self.band_mins[[3, 2, 1], None, None]
+            self.maxs = self.band_maxs[[3, 2, 1], None, None]
+        else:
+            self.mins = self.band_mins[:, None, None]
+            self.maxs = self.band_maxs[:, None, None]
+
+    def preprocess(self, sample: Dict[str, Any]) -> Dict[str, Any]:
+        """Transform a single sample from the Dataset."""
+        sample["image"] = sample["image"].float()
+        sample["image"] = (sample["image"] - self.mins) / (self.maxs - self.mins)
+        sample["image"] = torch.clip(  # type: ignore[attr-defined]
+            sample["image"], min=0.0, max=1.0
+        )
+        return sample
+
+    def prepare_data(self) -> None:
+        """Make sure that the dataset is downloaded.
+
+        This method is only called once per run.
+        """
+        OSCD(self.root_dir, split="train", bands=self.bands, checksum=False)
```

Comment:
I'm not sure how useful this is since `setup` also checks the same thing.

Reply:
I think we should make the datamodules download the dataset if it doesn't exist

Reply:
I think we should provide an option to do that, but it should be off by default. Principle of least surprise and whatnot.